src/Entities/DreamStuff/DreamDashCollider.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Microsoft.Xna.Framework;
+using Mono.Cecil.Cil;
+using Monocle;
+using MonoMod.Cil;
+using MonoMod.Utils;
+using System;
+
+namespace Celeste.Mod.CommunalHelper.Entities {
+    [Tracked]
+    class DreamDashCollider : Component {
+        public static readonly Color ActiveColor = Color.Teal;
+        public static readonly Color InactiveColor = Calc.HexToColor("044f63"); // darker teal
+
+        public Collider Collider;
+        public DreamBlockDummy Dummy;
+
+        public Action<Player> OnExit;
+
+        public DreamDashCollider(Collider collider, Action<Player> onExit_player_dreamJumped = null)
```

Comment:
`onExit_player_dreamJumped` is a very long parameter name that seems to imply it will only be invoked if the player dreamjumps? maybe shorten the name and explain it in an xmldoc comment instead

Reply:
I was going to make the delegate require a bool that is `true` when the player dreamjumps, but then I found out that you could dreamjump after it was called, so I removed it